Kickstarter-iOS/Views/Controllers/ProjectPageViewController.swift

```diff
@@ -0,0 +1,415 @@
+      self.pledgeCTAContainerView.rightAnchor.constraint(equalTo: self.view.rightAnchor),
+      self.pledgeCTAContainerView.bottomAnchor.constraint(equalTo: self.view.bottomAnchor)
+    ]
+
+    NSLayoutConstraint.activate(pledgeCTAContainerViewConstraints)
+  }
+
+  // MARK: Deinitialize
+
+  deinit {
+    self.sessionEndedObserver.doIfSome(NotificationCenter.default.removeObserver)
+    self.sessionStartedObserver.doIfSome(NotificationCenter.default.removeObserver)
+  }
+
+  // MARK: - Styles
+
+  public override func bindStyles() {
+    super.bindStyles()
+
+    _ = self
+      |> baseControllerStyle()
+  }
+
+  public override func bindViewModel() {
+    super.bindViewModel()
+
+    self.bindProjectPageViewModel()
+    self.bindSharingViewModel()
+    self.bindWatchViewModel()
+  }
+
+  // MARK: Orientation Change
+
+  public override func willTransition(
+    to newCollection: UITraitCollection,
+    with _: UIViewControllerTransitionCoordinator
+  ) {
+    self.viewModel.inputs.willTransition(toNewCollection: newCollection)
+  }
+
+  // MARK: - Private Helpers
+
+  private func bindProjectPageViewModel() {
+    self.viewModel.outputs.goToRewards
+      .observeForControllerAction()
+      .observeValues { [weak self] params in
+        let (project, refTag) = params
+
+        self?.goToRewards(project: project, refTag: refTag)
+      }
+
+    self.viewModel.outputs.goToManagePledge
+      .observeForControllerAction()
+      .observeValues { [weak self] params in
+        self?.goToManagePledge(params: params)
+      }
+
+    self.viewModel.outputs.configureChildViewControllersWithProject
+      .observeForUI()
+      .observeValues { [weak self] project, _ in
+        /** FIXME: How we do this might change in https://kickstarter.atlassian.net/browse/NTV-195
+         self?.contentController?.configureWith(value: (project, refTag))
+         */
+        self?.shareViewModel.inputs.configureWith(shareContext: .project(project), shareContextView: nil)
+        self?.watchProjectViewModel.inputs
+          .configure(with: (project, KSRAnalytics.PageContext.projectPage, nil))
+      }
+
+    self.viewModel.outputs.configurePledgeCTAView
+      .observeForUI()
+      .observeValues { [weak self] value in
+        self?.pledgeCTAContainerView.configureWith(value: value)
+      }
+
+    self.viewModel.outputs.dismissManagePledgeAndShowMessageBannerWithMessage
+      .observeForControllerAction()
+      .observeValues { [weak self] message in
+        self?.dismiss(animated: true, completion: {
+          self?.messageBannerViewController?.showBanner(with: .success, message: message)
+        })
+      }
+
+    self.viewModel.outputs.popToRootViewController
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.navigationController?.popToRootViewController(animated: false)
+      }
+  }
+
+  private func bindSharingViewModel() {
+    self.shareViewModel.outputs.showShareSheet
```

Comment:
Is the `ProjectNavBarViewController`and view model not reusable with delegation to this view controller? A little concerned about the bloat of three view models in one view controller.